### ISSUE TITLE
Request: Public key view and verifying

### ISSUE THREAD

**Issue by mrbiber** [PeerioTechnologies/peerio-client#24]
For the very security-conscious, it would be great if one could see the actual public key of a contact and if one could toggle a status, such as "verified" when one has compared public keys out-of-band, such that this status is automatically reset whenever the public key changes on the server.

**Comment by hdf** [comment]
+1

Also, really like that avatar generation idea, would be nice if this sort of thing got standardized in other projects. :)
( https://github.com/PeerioTechnologies/peerio-client/blob/master/src/chrome/js/peerio/crypto.js#L147 )

**Comment by mrbiber** [comment]
The capability to see the full key and to mark which keys one has verified is essential to effectively combat MITM attacks, without requiring users to constantly check that the keys are still the same. Alternatively, Peerio could display an alert whenever the server gives out a public key for a user that differs from a locally cached copy.

The avatar is a great addition, but it doesn't guarantee that key changes are obvious (and it's harder to export, compare for minor changes, and to compare out of band, like over the phone, as opposed to a text key). As things are now, one has to trust the server to not give out false public keys.

**Comment by hdf** [comment]
If marking keys as verified is implemented, I think it should be client side only, but where to store this information? It should be encrypted, but with what? Seems to stick out of the general design pattern.

**Comment by mrbiber** [comment]
Why can't the list of verified keys not be stored on the server, first signed and then encrypted with the user's private key? This way, the server operater can neither see nor manipulate information about which keys should count as verified.

**Comment by hdf** [comment]
Ok, I guess, that could work.

**Comment by kaepora** [comment]
Yes, we should absolutely have an option to display your Peerio miniLock ID. Expect this to be implemented very soon.

Thanks for the discussion regarding this necessary feature, everyone.

**Comment by IpsmLorem** [comment]
More than the display of your contact minilock ID (which is a must for side text channel verification), the feature is to store the verified status and reset it as soon as the identities change.

**Comment by mwfarb** [comment]
The SafeSlinger project provides a secret key exchange protocol that proves each user in a synchronous session has verified a key out of band (https://github.com/SafeSlingerProject/SafeSlinger-Media/wiki). We're also building a JavaScript version of the same (https://github.com/SafeSlingerProject/exchange-javascript). We'd be happy to collaborate.

**Comment by IpsmLorem** [comment]
@mwfarb,SafeSlinger is a nice feature, which deserve their own bug entry.

For MinilockID verification, maybe a display in a base 36 [0-9A-Z] encoding and with 3 digits grouping would be nice for verification by an audio channel (phone etc.)

**Comment by tenmalin** [comment]
I wonder MinilockID is sort of an identifier for a group to recognize the group association they create. Or this ID may have another meaning to the group? 

PS. I am also working on SafeSlinger project.

**Comment by SafeSlingerProject** [comment]
@tenmalin I'll reply on #132.

**Comment by SafeSlingerProject** [comment]
@tenmalin MiniLock Ids are roughly equivalent to at PGP key fingerprint I believe, each one representing an identity held by a person via remembered secret.

**Comment by IpsmLorem** [comment]
@SafeSlingerProject MiniLockID is the public key itself. Not just a fingerprint.

**Comment by IpsmLorem** [comment]
@kaepora the issue is not closed. Th epoint was to display the minilockID but to be able to remeber a verified status also. So the user can be able to detect a minilockID change.

**Comment by skylarnagao** [comment]
Reopening for request to offer toggle-set verification status.

Note, Peerio is designed to alert users if miniLock ID changes, will also change crypto-avatar.
